content/post/2018-07-01-Automate-your-grammar-checks.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Automate your grammar checks"
+date: 2018-07-01
+tags: ["documentation", "ci"]
+---
+
+# Continuous Integration for a spoken language
+
+I wanted to start a blog, so like a lot of people nowadays I've created a [GitHub repository](https://github.com/paulfantom/pawel.krupa.net.pl), added automatic deployment via [netlify](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/), and voila! But I still needed to write something... And while doing so it would be nice to use proper spelling, so why not adapt some kind of automatic spell checker?
+
+## Aspell? Hunspell? *spell?
+
+First of all it would be nice to find a CLI tool which could spell check plain-text files. A minute with google reveals that most frequently used spell checker tools are **Aspell** and **Hunspell**. They are similar in use and there are lots of articles on how to use them ([1](https://opensource.com/article/18/2/how-check-spelling-linux-command-line-aspell), [2](https://alexwlchan.net/2016/09/please-use-aspell/)). One problem is that they are primarily used in interactive mode and I would like to do spell checking in non-interactive mode, which produces [output unfriendly to human](https://github.com/hunspell/hunspell/blob/master/docs/hunspell.1.md#pipe-mode). Nonetheless I decided to use Hunspell in a bash wrapper for some more readable output.
```

Comment:
One problem is that they are primarily used in interactive mode. I would like to do spell-checking in non-interactive mode though and that produces not human-friendly output.